metrics/firebase/test/firestore/rules/user-profiles-test.js

```diff
@@ -0,0 +1,387 @@
+const async = require('async');
+const {
+  setupTestDatabaseWith,
+  getApplicationWith,
+  tearDown,
+} = require("./test_utils/test-app-utils");
+const { assertFails, assertSucceeds } = require("@firebase/testing");
+const {
+  passwordSignInProviderId,
+  googleSignInProviderId,
+  getAllowedEmailUser,
+  getDeniedEmailUser,
+  userProfiles,
+  getUserProfile,
+  allowedEmailDomains,
+} = require("./test_utils/test-data");
+
+describe("", async () => {
+  const collection = "user_profiles";
+  const superUserApp = await getApplicationWith(
+    getAllowedEmailUser(passwordSignInProviderId, true, '1')
+  )
+
+  const signInProviders = [
+    {
+      'title': 'password',
+      'description': 'Authenticated with a password and ',
+      'users': [
+        {
+          'description': 'allowed email domain user who is not an owner of the user profile with not a verified email',
+          'app': await getApplicationWith(
+            getAllowedEmailUser(passwordSignInProviderId, false)
+          ),
+          'can': {
+            'create': true,
+            'read': false,
+            'readSelf': false,
```

Comment:
```suggestion
            'readOwn': false,
```
Would it be more clear?